### ISSUE TITLE
loop check partition number coroutine will exit stop to consume message without return the problem to the main routine

### ISSUE THREAD

**Issue by bestbug456** [IBM/sarama#1685]
##### Versions

*Please specify real version numbers or git SHAs, not just "Latest" since that changes fairly regularly.*

| Sarama | Kafka | Go |
|--------|-------|----|
| v1.26.1 | 0.11 | 0.14.1 |

##### Configuration

What configuration values are you using for Sarama and Kafka?
``` go
	config.Consumer.Group.Rebalance.Strategy = sarama.BalanceStrategyRange
	config.Consumer.Return.Errors = true
	config.Consumer.Group.Session.Timeout = 30 * time.Second
	config.Consumer.Offsets.Initial = sarama.OffsetNewest
	config.Consumer.Group.Rebalance.Timeout = 120 * time.Second
```

##### Logs

When filing an issue please provide logs from Sarama and Kafka if at all
possible. You can set `sarama.Logger` to a `log.Logger` to capture Sarama debug
output.

<details><summary>logs: CLICK ME</summary>
<p>

```
[sarama] 2020/04/30 10:15:44 loop check partition number coroutine will exit, topics [my-topic-name]
[sarama] 2020/04/30 10:15:44 consumer/broker/2 closed dead subscription to my-topic-name/2
[sarama] 2020/04/30 10:15:44 consumer/broker/2 closed dead subscription to my-topic-name/8
[sarama] 2020/04/30 10:15:44 consumer/broker/2 closed dead subscription to my-topic-name/5
[sarama] 2020/04/30 10:15:44 consumer/broker/2 closed dead subscription to my-topic-name/11
[sarama] 2020/04/30 10:15:44 consumer/broker/1 closed dead subscription to my-topic-name/1
[sarama] 2020/04/30 10:15:44 consumer/broker/1 closed dead subscription to my-topic-name/10
[sarama] 2020/04/30 10:15:44 consumer/broker/1 closed dead subscription to my-topic-name/4
[sarama] 2020/04/30 10:15:44 consumer/broker/1 closed dead subscription to my-topic-name/7
[sarama] 2020/04/30 10:15:44 consumer/broker/0 closed dead subscription to my-topic-name/0
[sarama] 2020/04/30 10:15:44 consumer/broker/0 closed dead subscription to my-topic-name/6
[sarama] 2020/04/30 10:15:44 consumer/broker/0 closed dead subscription to my-topic-name/3
[sarama] 2020/04/30 10:15:44 consumer/broker/0 closed dead subscription to my-topic-name/9
```

</p>
</details>

##### Problem Description
Hello, we update the library to v1.26.1 and we notice that when the log `loop check partition number coroutine will exit` is printed the library stop consuming messaging and stop retry to reconnect to Kafka but no error are reported to the channel. This behavior block us to adopt countermeasures and the microservices can't restore connection to kafka (after manual reboot the system return working for a while but then we see this logs again). We are setting something wrong or this is an intended behaviour? If it's so how can the main routine catch the error and fix it?

**Comment by WMLHUST** [comment]
I have the same problem, exactly the same.

**Comment by andrelu** [comment]
I Have the same issue consuming from kafka in Amazon MSK.

**Comment by d1egoaz** [comment]
friends, how the consumer code looks like?

are you doing this:
https://github.com/Shopify/sarama/blob/master/consumer_group.go#L41-L44

https://github.com/Shopify/sarama/blob/master/examples/consumergroup/main.go#L101-L113

tl;dr it might not be an issue at all, it might be just  `Consume` needed to be called agains as something have changed on the server/client and a rebalance was triggered

**Comment by bestbug456** [comment]
Hello @d1egoaz I think you just caught the problem. So your suggestion is to change and not call the function consume once but each time we consume something? If we want to optimise the call to your library how can we know when we have to do this reboot?

**Comment by d1egoaz** [comment]
@bestbug456 
could you please expand on this a little bit (some code maybe?), as I didn't get it
>  and not call the function consume once but each time we consume something


I'm going to try to explain a little bit better

This function call will block until something like the following happens:
![image](https://user-images.githubusercontent.com/950087/86270889-d6ab9b00-bb80-11ea-8540-ad26cba1cfba.png)

- an error has occurred: and the `err` variable would have the error, you can choose to continue or close the process, most of the time if there is an error, you should log it and exit, or might want to recreate entirely the client.
- it will stop consuming, and no error is returned, most of the time it's because a rebalance was triggered, so you'll need to call `Consume` again, as all the dance for `Setup`, `ConsumeClaim` needs to be re-done, as this member might have different (more or less or different) replicas (claims) to consume, usually when a new member joins the consumer group, all the current members will need to rebalance, so that's why is important to use an infinite for loop here
![image](https://user-images.githubusercontent.com/950087/86271334-8f71da00-bb81-11ea-9e4e-0a04bd371f32.png)
